### PR TITLE
fix various typos

### PR DESCRIPTION
**Pull request by matthiaskrgr**

### CODE REVIEW

File: src/libcore/tests/num/dec2flt/mod.rs [hunk 1]
```diff
@@ -44,7 +44,7 @@ fn ordinary() {
 #[test]
 fn special_code_paths() {
     test_literal!(36893488147419103229.0); // 2^65 - 3, triggers half-to-even with even significand
-    test_literal!(101e-33); // Triggers the tricky underflow case in AlgorithmM (for f32)
+    test_literal!(101e-33); // Triggers the tricky underflow case in algorithm (for f32)
```
Comment (petrochenkov) [inline src/libcore/tests/num/dec2flt/mod.rs]: Not a typo, `AlgorithmM` is an algorithm name.

File: src/librustc_expand/proc_macro.rs [hunk 2]
```diff
@@ -133,7 +133,7 @@ impl MultiItemModifier for ProcMacroDerive {
         };
 
         let error_count_before = ecx.parse_sess.span_diagnostic.err_count();
-        let msg = "proc-macro derive produced unparseable tokens";
+        let msg = "proc-macro derive produced unparsable tokens";
```
Comment (petrochenkov) [inline src/librustc_expand/proc_macro.rs]: This looks weird to me and Google search gives 3.5 times more hits for "unparseable" than for "unparsable".

File: src/libcore/tests/num/dec2flt/mod.rs (hunk 1 above)
Comment (matthiaskrgr) [reply src/libcore/tests/num/dec2flt/mod.rs]: Thanks, reverted.

File: src/librustc_expand/proc_macro.rs (hunk 2 above)
Comment (matthiaskrgr) [reply src/librustc_expand/proc_macro.rs]: reverted as well (including corresponding tests)

**Pull request merged**